caikit/runtime/client/utils.py

```diff
@@ -31,8 +33,45 @@ def construct_grpc_channel(
     target: str,
     options: Optional[List[Tuple[str, str]]] = None,
     tls: Optional[ConnectionTlsInfo] = None,
+    retries: Optional[int] = None,
+    retry_options: Optional[Dict[str, str]] = None,
 ) -> grpc.Channel:
     """Helper function to construct a grpc Channel with the given TLS config"""
+    # Add retry option if one was provided
+    if retries and retries > 1:
+        options.append(("grpc.enable_retries", 1))
+
+        # Only add service_config if it wasn't already added to the GRPC option
+        # this stops us from overriding an advanced config
+        options_contain_service_config = False
+        for option_name, _ in options:
+            if option_name == "grpc.service_config":
+                options_contain_service_config = True
+                break
+
+        if not options_contain_service_config:
+            service_config = {
+                "methodConfig": [
+                    {
+                        "name": [{}],
+                        "retryPolicy": {
+                            "maxAttempts": retries,
+                            "initialBackoff": "0.1s",
```

Comment:
It looks like this is hard-coded here, but overridable if the user provides these advanced configs in `options`, right?

Reply:
Yep exactly! Anything the user provides in `retry_options` will replace any of these static keys. Including `maxAttempts` if they supply it